controllers/grid/navigationMenus/form/PKPNavigationMenuItemsForm.inc.php

```diff
@@ -95,38 +95,53 @@ function fetch($request, $template = 'controllers/grid/navigationMenus/form/navi
 			}
 		}
 
-		$templateMgr->assign(array(
+		$templateArray = array(
 			'navigationMenuItemTypeTitles' => $typeTitles,
 			'navigationMenuItemTypeDescriptions' => json_encode($typeDescriptions),
 			'navigationMenuItemTypeConditionalWarnings' => json_encode($typeConditionalWarnings),
-		));
+		);
+
+		$templateMgr->assign($templateArray);
 
 		return parent::fetch($request, $template, $display);
 	}
 
 	/**
 	 * Initialize form data from current navigation menu item.
 	 */
-	function initData() {
+	function initData($data = array()) {
 		$navigationMenuItemDao = DAORegistry::getDAO('NavigationMenuItemDAO');
 		$navigationMenuItem = $navigationMenuItemDao->getById($this->navigationMenuItemId);
 
 		if ($navigationMenuItem) {
-			$this->_data = array(
+			$formData = array(
 				'path' => $navigationMenuItem->getPath(),
 				'title' => $navigationMenuItem->getTitle(null),
 				'url' => $navigationMenuItem->getUrl(),
 				'menuItemType' => $navigationMenuItem->getType(),
 			);
+
+			$this->_data =  array_merge($data, $formData);
+
 			$this->setData('content', $navigationMenuItem->getContent(null)); // Localized
-		} 
+		}
 	}
 
 	/**
 	 * Assign form data to user-submitted data.
 	 */
 	function readInputData() {
-		$this->readUserVars(array('navigationMenuItemId', 'content', 'title', 'path', 'url','menuItemType'));
+		$formInputData = array('navigationMenuItemId', 'path', 'content', 'title', 'url','menuItemType');
+
+		$this->readUserVars($formInputData);
+	}
+
+	/**
+	 * @copydoc Form::getLocaleFieldNames()
+	 */
+	function getLocaleFieldNames() {
+		$dao = DAORegistry::getDAO('NavigationMenuItemDAO');
+		return $dao->getLocaleFieldNames();
```

Comment:
It's a minor thing, but we usually name the `$dao` variable according to the DAO. So in this case it would be `$navigationMenuItemDao = ...`.

(Alternatively, you could just do `DAORegistry::getDAO('NavigationMenuItemDAO')->getLocaleFieldNames()` too.)